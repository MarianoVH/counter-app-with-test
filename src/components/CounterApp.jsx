import React, { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const addValue = () => setCounter(counter + 1);
  const subtractValue = () => setCounter(counter - 1);
  const reset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={addValue}>+1</button>
      <button onClick={reset}>reset</button>
      <button onClick={subtractValue}>-1</button>
    </>
  );
};

CounterApp.propType = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 0,
};
