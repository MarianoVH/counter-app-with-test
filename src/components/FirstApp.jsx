import React from "react";
import PropTypes from "prop-types";

export const FirstApp = ({ subTitle, title }) => (
  <>
    <h1>{title}</h1>
    <h2>{subTitle}</h2>
  </>
);

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  subTitle: "I'm a subtitle",
};
