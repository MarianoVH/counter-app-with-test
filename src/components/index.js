export * from "./CounterApp";
export * from "./FirstApp";
