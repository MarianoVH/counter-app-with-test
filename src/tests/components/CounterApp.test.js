import React from "react";
import { shallow } from "enzyme";

import { CounterApp } from "../../components";

describe("Testing CounterApp component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });
  test("check render with snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("check props", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("100");
  });

  test("check button increment", () => {
    wrapper.find("button").at(0).simulate("click");

    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("1");
  });

  test("check button reset", () => {
    wrapper.find("button").at(1).simulate("click");

    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("0");
  });

  test("check button decrement", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");

    const counterText = wrapper.find("h2").text();

    expect(counterText).toBe("100");
  });
});
