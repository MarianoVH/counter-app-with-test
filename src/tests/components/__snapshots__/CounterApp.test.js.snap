// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CounterApp component check render with snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
